godot-core/src/obj/dyn_gd.rs

```diff
@@ -137,6 +137,70 @@ use std::{fmt, ops};
 /// and it can query the dynamic type of the object. Based on that type, it can find the `impl Health` implementation matching the correct class.
 /// Behind the scenes, everything is wired up correctly so that you can restore the original `DynGd` even after it has passed through Godot.
 ///
+/// # Type inference
+///
+/// If a class implements more than one `AsDyn<D>` relation (usually via `#[godot_dyn]`), type inference will only work when the trait
+/// used for `D` explicitly declares a `: 'static` bound.
```

Comment:
This section could probably come last, no?

Exporting is definitely more important (arguably also more than dyn-re-enrichment).

Reply:
hmm, I think you are right – albeit I'm finding _Polymorphic `dyn` re-enrichment_ a bit more important :thinking: (reasoning – the first two sections explain how to use it and how it works, while following ones focus on documenting various quirks)

I put `Type inference` as the last section